.github/actions/verify-token/src/validate.ts

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2022 SLSA Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WIHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { githubObj } from "../src/types";
+
+export function validateGitHubFields(gho: githubObj): void {
+  // actor
+  validateFieldNonEmpty("github.actor", gho.actor);
```

Comment:
maybe you could have the 2 functions `validateFieldNonEmpty` and `validateField` in a single function, like `validateFieldEqualNonEmpty`

Reply:
`validateField` and `validateFieldNonEmpty` were used elsewhere and I didn't want to change that logic. Though I think that `validateField` always wants to check non-empty as well so maybe adding that check is ok.